cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -78,6 +78,9 @@ DrawableSubmap::DrawableSubmap(const int trajectory_id, const int submap_index,
 }
 
 DrawableSubmap::~DrawableSubmap() {
+  ::cartographer::common::MutexLocker locker(&mutex_);
+  quitting_ = true;
+  locker.Await([this]() REQUIRES(mutex_) { return !query_in_progress_; });
```

Comment:
Seems line 81 and 83 fix the issue and provide the safety mentioned in the commit description. Keep only these?